### PR TITLE
Modified ZooTabletStateStore to use TabletManagementIterator

### LINKED ISSUE #3587: Attempt to use TabletManagementIterator for root tablet
The root tablet does not use the TabletManagementIterator to make decisions about the root tablet for the TabletGroupWatcher.  Its specialized code is here : 

https://github.com/apache/accumulo/blob/406552e46df4412cf8f4f52962446773a9eabee5/server/base/src/main/java/org/apache/accumulo/server/manager/state/ZooTabletStateStore.java#L74-L86

Adapting this to use TabletManagementIterator in some way would be nice as it would make the root tablets behavior less specialized and more likely to behave like the rest of accumulo.  May be able to use code similar to the following

https://github.com/apache/accumulo/blob/406552e46df4412cf8f4f52962446773a9eabee5/server/base/src/main/java/org/apache/accumulo/server/metadata/RootConditionalWriter.java#L104

to obtain an Accumulo iterator over the root tablet metadata that could be the source iterator for TabletManagementIterator.

### PR DESCRIPTION
**Pull request by dlmarion**
Closes #3587

### CODE REVIEW

File: server/base/src/main/java/org/apache/accumulo/server/manager/state/ZooTabletStateStore.java [hunk 1]
```diff
@@ -67,50 +73,78 @@ public ClosableIterator<TabletManagement> iterator(List<Range> ranges,
...
+        }
 
-      @Override
-      public void close() {}
+        Key k = tmi.getTopKey();
+        Value v = tmi.getTopValue();
+        Map<Key,Value> m = Map.of(k, v);
+        Entry<Key,Value> e = m.entrySet().iterator().next();
+        try {
+          TabletManagement tm = TabletManagementIterator.decode(e);
```
Comment (keith-turner) [inline server/base/src/main/java/org/apache/accumulo/server/manager/state/ZooTabletStateStore.java:133]: Could drop `returnedRootTabletMetadata` variable if the following lines were added.

```suggestion
          tmi.next();
          Preconditions.checkState(!tmi.hasTop(), "Saw multiple tablet metadata entries for root table");
          TabletManagement tm = TabletManagementIterator.decode(e);
```

File: server/base/src/main/java/org/apache/accumulo/server/manager/state/ZooTabletStateStore.java [hunk 2]
```diff
@@ -67,50 +73,78 @@ public ClosableIterator<TabletManagement> iterator(List<Range> ranges,
...
-
+        return result;
       }
 
       @Override
-      public void remove() {
-        throw new UnsupportedOperationException();
-      }
+      public TabletManagement next() {
+        if (closed.get()) {
```
[the remark this answers is not in the document]
Comment (dlmarion) [reply server/base/src/main/java/org/apache/accumulo/server/manager/state/ZooTabletStateStore.java]: Implemented in 85f67fe

File: server/base/src/main/java/org/apache/accumulo/server/manager/state/ZooTabletStateStore.java [hunk 3]
```diff
@@ -67,50 +73,78 @@ public ClosableIterator<TabletManagement> iterator(List<Range> ranges,
...
+          throw new NoSuchElementException(this.getClass().getSimpleName() + " is closed");
+        }
 
-      @Override
-      public void close() {}
+        Key k = tmi.getTopKey();
+        Value v = tmi.getTopValue();
+        Map<Key,Value> m = Map.of(k, v);
+        Entry<Key,Value> e = m.entrySet().iterator().next();
```
[the remark this answers is not in the document]
Comment (dlmarion) [reply server/base/src/main/java/org/apache/accumulo/server/manager/state/ZooTabletStateStore.java]: Implemented in 85f67fe

File: server/base/src/main/java/org/apache/accumulo/server/manager/state/ZooTabletStateStore.java (hunk 1 above)
Comment (dlmarion) [reply server/base/src/main/java/org/apache/accumulo/server/manager/state/ZooTabletStateStore.java:132]: Implemented in 85f67fe

**Pull request merged**